plugins/out_s3/s3.c

```diff
@@ -1243,6 +1259,245 @@ static struct multipart_upload *create_upload(struct flb_s3 *ctx,
     return m_upload;
 }
 
+/* Adds an entry to upload queue */
+static void add_to_queue(struct flb_s3 *ctx, struct s3_file *upload_file,
+                 struct multipart_upload *m_upload_file, const char *tag, int tag_len)
+{
+    int ret;
+    struct upload_queue *upload_contents;
+    char *tag_cpy;
+
+    if (upload_file == NULL && m_upload_file == NULL) {
+        flb_plg_error(ctx->ins, "Tried to add entry without file to upload_queue");
```

Comment:
this error message seems to indicate a problem in the code, rather than something that the user can understand and take action on... this likely means the code needs to be re-designed.... error messages should be useful to a user

Reply:
Whoops, just realized I check to make sure both aren't NULL outside the call of this function :)